api/v1alpha1/envoyproxy_types.go

```diff
@@ -255,6 +255,9 @@ const (
 
 	// EnvoyFilterRouter defines the Envoy HTTP router filter.
 	EnvoyFilterRouter EnvoyFilter = "envoy.filters.http.router"
+
```

Comment:
Since this is a new filter, I think you should pay attention to the `newOrderedHTTPFilter`, take https://github.com/envoyproxy/gateway/pull/5171 as example.